cva6/tests/custom/sv32/vm_misaligned_superpage.S

```diff
@@ -0,0 +1,207 @@
+    #ifdef smode
+      CHANGE_T0_S_MODE(a1)                                                  # changes mode M to S and set the MEPC
+    #else
+      CHANGE_T0_U_MODE(a1)                                                  # changes mode M to U and set the MEPC
+    #endif
+
+.align 12                                                                   # to make the address misliagned to test execute access
+check_execute:
+
+    li t1, 0x45                                                             # page fault should raise 
+    TEST_STATUS                                                             # checks the status of the test
+
+    #ifdef smode
+      SMODE_ECALL                                                           # changes mode M to S and set the MEPC
+    #else
+     UMODE_ECALL                                                            # changes mode M to U and set the MEPC
+    #endif
+
+test_pass:
+
+    li x1, 0                                                                # Write 0 in x1 if test pass
+    j exit                                                                  # Jump to exit
+
+test_fail:
+
+    li x1, 1                                                                # Write 1 in x1 if test failed
+    j exit                                                                  # Jump to exit
+                                                                   
+trap_handler:
+
+    csrr t0, mcause                                                         # read the value of mcause 
+    la t1, rvtest_check                                                     # load the address of trvtest_check
+    
+    lw t2, 0(t1)                                                            # if cause expected then load 1 else 0
+    lw t3, 4(t1)                                                            # load the expected value of mepc 
+    lw t4, 8(t1)                                                            # load the expected value of mcause  
+
+    li  t1, CAUSE_SUPERVISOR_ECALL                                          # load the value of supervisor ecall
+    beq t0,t1,next_instr                                                    # checks if ecall is occured
+
+    li  t1, CAUSE_USER_ECALL                                                # load the value of user ecall
+    beq t0,t1,next_instr                                                    # checks for ecall is occured
+
+    beqz t2, exit                                                           # Jumps to exit if cause is not expected
+ 
+    csrr t5,mepc                                                            # read the value of mepc 
+    bne t3,t5,exit                                                          # check the value of mepc with it's expected value
+
+    bne  t0, t4, exit                                                       # jumps to exit if EXPECTED_CAUSE is'nt equal to mcause
```

Comment:
```suggestion
    csrr t0, mcause                                                      # read the value of mcause 
    bne  t0, t4, exit                                                       # jumps to exit if EXPECTED_CAUSE isn't equal to mcause
```

Reply:
No further updates are needed for trap_handler based on our previous discussion